src/synthesizer/particle/stars.py

```diff
@@ -363,12 +362,38 @@ def generate_lnu(
             Numpy array of integrated spectra in units of (erg / s / Hz).
         """
 
-        # Ensure we have a total key in the grid. If not error.
+        # Ensure we have a key in the grid. If not error.
         if spectra_name not in list(grid.spectra.keys()):
             raise exceptions.MissingSpectraType(
                 "The Grid does not contain the key '%s'" % spectra_name
             )
 
+        # Check the fraction of particles outside of the grid (these will be
+        # pinned to the edge of the grid) by finding those inside
+        age_inside_mask = np.logical_and(
+            self.log10ages < grid.log10age[-1],
+            self.log10ages > grid.log10age[0],
+        )
+        met_inside_mask = np.logical_and(
+            self.metallicities < grid.metallicity[-1],
+            grid.metallicity[0] < self.metallicities,
+        )
+        inside_mask = np.logical_and(age_inside_mask, met_inside_mask)
+        n_out = self.metallicities[~inside_mask].size
+        ratio_out = n_out / self.nparticles
+
+        # Tell the user if there are particles outside the grid, or worst case
+        # error if too many are outside
+        if ratio_out > 0.5:
+            raise exceptions.GridError(
```

Comment:
still raising an exception here?

Reply:
🤦 sorry, I only pushed the change to the particle version. Pushed both now

Reply:
Pushed now (had to also handle a local conflict because of the merge I did here).